### PR TITLE
Crude port of Tofino dynamic hash to C++

### PR DESCRIPTION
**Pull request by asl**

### CODE REVIEW

**Comment by asl** [comment]
Spin-off from https://github.com/p4lang/p4c/pull/5033

**Review by fruffy — APPROVED** [review]
Thanks! We can also remove that unnecessary split into include/src since we do not do that in the rest of the code. I can also take care of that later.

File: backends/tofino/bf-p4c/CMakeLists.txt [hunk 1]
```diff
@@ -970,7 +970,7 @@ target_include_directories(
   # FIXME: This should be pulled with FetchContent.
   PRIVATE "${BFN_P4C_SOURCE_DIR}/third_party/spdlog/include"
 )
-add_dependencies(tofinobackend ir-generated frontend genLogging bfn_p4runtime)
+add_dependencies(tofinobackend frontend genLogging bfn_p4runtime)
```
Comment (fruffy) [inline backends/tofino/bf-p4c/CMakeLists.txt:973]: Assume this is remove because of the frontend dependency?

Comment (asl) [reply backends/tofino/bf-p4c/CMakeLists.txt:973]: Yes. Actually, the majority `add_dependencies` are not needed at all. This seems to be an old hack used to "close" some absence of proper dependencies. This needs to be eventually removed from everywhere and replaces with proper deps (e.g. `target_link_libraries`, etc.)

**Pull request merged**